dxr/plugins/python/indexers.py

```diff
@@ -249,44 +257,43 @@ def analyze_tokens(self):
             if tok_type == token.NAME:
                 # AST nodes for classes and functions point to the position of
                 # their 'def' and 'class' tokens. To get the position of their
-                # names, we look for 'def' and 'class' tokens and store the
-                # position of the token immediately following them.
-                if node_start and node_type == 'definition':
-                    node_start_table[node_start[0]] = (start, end)
-                    node_type, node_start = None, None
-                    continue
-
-                if tok_name in ('def', 'class'):
-                    node_type, node_start = 'definition', (utf8_start, start)
-                    continue
-
-                # Record all name nodes in the token table.  Currently unused,
-                # but will be needed for recording variable references.
-                node_start_table[utf8_start] = (start, end)
-                node_type, node_start = 'name', (utf8_start, start)
+                # names, we start the queue for the current parenthesis level
+                # at the byte offset for the keyword token, but only start
+                # pushing character offsets once we're past the keyword.
+                paren_stack.setdefault(paren_level, (utf8_start, []))
+
+                if tok_name not in ('def', 'class'):
+                    paren_stack[paren_level][1].append((start, end))
```

Comment:
We could be a little more efficient (saving a lookup) by saving the result of the `setdefault` just above and then saying `saved_thingy[1].append(...)` here.